pkg/apis/ark/v1/backup_locations.go

```diff
@@ -0,0 +1,89 @@
+/*
+Copyright 2018 the Heptio Ark contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// BackupStorageLocation is a location where Ark stores backup objects.
+type BackupStorageLocation struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+
+	Spec   BackupStorageLocationSpec   `json:"spec"`
+	Status BackupStorageLocationStatus `json:"status"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// BackupStorageLocationList is a list of BackupStorageLocations.
+type BackupStorageLocationList struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+	Items             []BackupStorageLocation `json:"items"`
+}
+
+// ObjectStorageLocation specifies the settings necessary to connect to a provider's object storage.
+type ObjectStorageLocation struct {
+	// Provider is the cloud provider where object storage resides.
+	Provider string `json:"provider"`
```

Comment:
Do you think `Provider` should be a field directly on `spec`, or under `ObjectStorageLocation`? I'm not sure, but thinking about scenarios where we add additional location types (e.g. git).  Same question for `Config`

Reply:
I think that makes sense. I'll update it.